apps/server/lib/lexical/server/window.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Lexical.Server.Window do
+  alias Lexical.Protocol.Notifications.LogMessage
+  alias Lexical.Protocol.Notifications.ShowMessage
+  alias Lexical.Server.Transport
+
+  @type level :: :error | :warning | :info | :log
+
+  @spec log(level, String.t(), [{:label, String.t()}]) :: String.t()
+  def log(level, message, opts \\ [])
+
+  def log(level, message, opts) when level in [:error, :warning, :info, :log] do
```

Comment:
Thoughts on generating helper functions for each level?

Reply:
like...
```elixir
@levels [:error, :warning, :info]

for level <- levels do 
  def unquote(level)(message, opts \\ []) do 
    log(level, message, opts)
   end
end
```

Reply:
Love it!
<img width="556" alt="Screenshot 2023-10-21 at 12 22 51 PM" src="https://github.com/lexical-lsp/lexical/assets/51095001/093b409c-1247-4d85-b81d-4200fdc7ca34">